utilities/Microsoft.Quantum.Katas/CheckKataMagic.cs

```diff
@@ -0,0 +1,203 @@
+
+        /// <summary>
+        /// What this Magic does when triggered. It will:
+        /// - find the Kata to execute based on the Kata name,
+        /// - compile the code after found after the name as the user's answer.
+        /// - run (simulate) the kata.
+        /// </summary>
+        public virtual ExecutionResult Run(string input, IChannel channel)
+        {
+            channel = channel.WithNewLines();
+
+            // Expect exactly two arguments, the name of the Kata and the user's answer (code).
+            var args = input?.Split(new char[] { ' ', '\n', '\t' }, 2);
+            if (args == null || args.Length != 2)
+            {
+                channel.Stderr("Invalid parameters. Usage: `%kata Test \"Q# operation\"`");
+                return ExecuteStatus.Error.ToExecutionResult();
+            }
+
+            var name = args[0];
+            var code = args[1];
+
+            var kata = FindKata(name);
+            if (kata == null)
+            {
+                channel.Stderr($"Invalid test name: {name}");
+                return ExecuteStatus.Error.ToExecutionResult();
+            }
+
+            var userAnswer = Compile(code, channel);
+            if (userAnswer == null) { return ExecuteStatus.Error.ToExecutionResult(); }
+
+            return Simulate(kata, userAnswer, channel)
+                ? "Success!".ToExecutionResult()
+                : ExecuteStatus.Error.ToExecutionResult();
+        }
+
+        /// <summary>
+        /// Compiles the given code. Checks there is only one operation defined in the code,
+        /// and returns its corresponding OperationInfo
+        /// </summary>
+        public virtual string Compile(string code, IChannel channel)
+        {
+            try
+            {
+                var result = Compiler.IdentifyElements(code);
+
+                // Gets the names of all the operations found for this snippet
+                var opsNames =
+                    result
+                        .Where(e => e.IsQsCallable)
+                        .Select(e => e.ToFullName().WithoutNamespace(Microsoft.Quantum.IQSharp.Snippets.SNIPPETS_NAMESPACE))
+                        .OrderBy(o => o)
+                        .ToArray();
+
+                if (opsNames.Length > 1)
+                {
+                    channel.Stdout("Expecting only one Q# operation in code. Using the first one");
+                }
+
+                return opsNames.First();
+            }
+            catch (CompilationErrorsException c)
+            {
+                foreach (var m in c.Errors) channel.Stderr(m);
+                return null;
+            }
+            catch (Exception e)
+            {
+                Logger?.LogWarning(e, "Unexpected error.");
+                channel.Stderr(e.Message);
+                return null;
+            }
+        }
+
+        /// <summary>
+        /// Executes the given kata using the provided <c>userAnswer</c> as the actual answer.
```

Comment:
Same question - I suspect this refers to a single task, not a whole kata

Reply:
@anpaz-msft  same here - thanks!